types/api.d.ts

```diff
@@ -1,6 +1,8 @@
 import { Uri } from "vscode";
 import { WorkspaceApi } from "@sap/project-api";
 
+declare type BasWorkspaceApi = Pick<WorkspaceApi, "getProjects" | "getProjectUris" | "onWorkspaceChanged">;
```

Comment:
using Pick is better than Omit as we explicitly control what our API includes.
👍